File: src/server/src/main/java/com/project/domain/image/Image.java
package com.project.domain.image;

import com.project.domain.BaseEntity;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;

import java.util.UUID;

@MappedSuperclass
public abstract class Image extends BaseEntity {
    @Id
    private UUID id;
    private String url;
    protected Image() {
    }

    protected Image(String url) {
        super();
        this.id = UUID.randomUUID();
        this.url = url;
    }

    public UUID getId() {
        return id;
    }

    public String getUrl() {
        return url;
    }
}

File: src/server/src/main/kotlin/com/project/domain/image/Image.kt
package com.project.domain.image

import com.project.domain.BaseEntity
import jakarta.persistence.Id
import jakarta.persistence.MappedSuperclass
import java.util.UUID

@MappedSuperclass
abstract class Image protected constructor() : BaseEntity() {
    @Id
    val id: UUID = UUID.randomUUID()
    var url: String? = null
        private set


    protected constructor(url: String?) : this() {
        this.url = url
    }
}
